src/main/kotlin/deadline/application/CreateDeadline.kt

```diff
@@ -0,0 +1,22 @@
+package deadline.application
+
+import deadline.domain.Deadline
+import deadline.domain.DeadlineRepository
+import deadline.domain.Time
+import note.domain.NoteIdentifier
+
+class AlreadyConfiguredDeadline : Throwable()
+
+class CreateDeadline(private val repository: DeadlineRepository) {
+
+    /***
+     * Creates a new deadline and saves it to the DeadlineRepository
+     * @throws AlreadyConfiguredDeadline if a new deadline already exists
+     * @return The deadline instance that was saved inside the repository
+     */
+    fun create(id: NoteIdentifier, time: Time): Deadline {
```

Comment:
This shouldn't receive domain objects, but primitives mister worlwide.

Reply:
Who told you that? Oh, wait....

![Whoami](https://media.giphy.com/media/l41m4sEBkOvMr3hL2/giphy.gif)

Jokes aside, yes, I am going to change that asap.